src/map/skill.c

```diff
@@ -7306,27 +7306,37 @@ static int skill_castend_nodamage_id(struct block_list *src, struct block_list *
 			break;
 
 		case PR_STRECOVERY:
-			if(status->isimmune(bl)) {
-				clif->skill_nodamage(src,bl,skill_id,skill_lv,0);
+			if (status->isimmune(bl) != 0) {
+				clif->skill_nodamage(src, bl, skill_id, skill_lv, 0);
 				break;
 			}
-			if (tsc && tsc->opt1) {
-				status_change_end(bl, SC_FREEZE, INVALID_TIMER);
-				status_change_end(bl, SC_STONE, INVALID_TIMER);
-				status_change_end(bl, SC_SLEEP, INVALID_TIMER);
-				status_change_end(bl, SC_STUN, INVALID_TIMER);
-				status_change_end(bl, SC_WHITEIMPRISON, INVALID_TIMER);
+
+			if (bl->type == BL_PC && tstatus->def_ele != ELE_UNDEAD) {
+				if (tsc != NULL && tsc->opt1 > 0) {
```

Comment:
```tsc->opt1 != 0```
please keep translate the other behavior as it was written before, also it's a bitmask variable which makes `> 0` incorrect.

Reply:
Since it's an unsigned integer, `> 0` doesn't make more or less sense than `!= 0` to me, even  if it's a bitmask variable. Correct me if I'm wrong.

> please keep translate the old behavior as it was written before

I should check implicitly even though it's not a boolean? This is how it was written before I changed it: `if (tsc && tsc->opt1) {`
Maybe I just misunderstood you.

Reply:
`>` `<` `+` `-` operations are invalid operations on bitmask fields, therefore your `> 0` give wrong impression.
and by translating old behavior it means
``` if (tsc->opt1) ```
is true whenever opt1 is not zero, therefore when you fix the coding style you should preserve the old behavior `!= 0` unless you're explicitly fixing a behavior. which is not the case here anyway.

Reply:
Ah, okay. Thanks for the explanation. :+1:

Reply:
Replaced `tsc->opt1 > 0` with `tsc->opt1 != 0`.